### PR TITLE
Versioning integrity - Create readme files for all versions of crayons

### PR DESCRIPTION
**Pull request by srivalli-sivaramasankaran**
## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My commits have standard messages as mentioned in [Contributing Guidelines](./../blob/next/CONTRIBUTING.md)

### CODE REVIEW

File: www/.vuepress/config.js [hunk 1]
```diff
@@ -15,6 +15,34 @@ const customObjectsComponents = components.filter((comp) =>
...
+  console.log('version.json fetch error ', err);
+  versionArr = [];
+}
+versionsArr.forEach((v) => {
+  v['key'] === '2.13.4' && versionMap['v2Components'].push(`versions/v${v['key']?.split('.')[0]}.x/${v.key}/`);
+  v['key'].startsWith('3.') && !v['key'].includes('beta') && versionMap['v3Components'].push(`versions/v${v['key']?.split('.')[0]}.x/${v.key}/`);
+  v['key'].startsWith('4.') && versionMap['v4Components'].push(`versions/v${v['key']?.split('.')[0]}.x/${v.key}/`);
+});
+
+console.log('versionMap-->', versionMap)
```
Comment (arvindanta) [inline www/.vuepress/config.js:33]: kindly remove logs

File: www/scripts/template.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+<script
+  type="module"
+  src="node_modules/@freshworks/crayons/dist/crayons/crayons.esm.js"
+></script>
+```
+
+Documentation of all crayons versions can be found [here](/versions/) .
+
+
+<!-- {{#css}}
```
Comment (arvindanta) [inline www/scripts/template.md:62]: can you uncomment the usage of css utils via node_modules and check

File: www/.vuepress/config.js [hunk 3]
```diff
@@ -15,6 +15,34 @@ const customObjectsComponents = components.filter((comp) =>
   comp.startsWith('components/crayons-custom-objects/')
 );
 
+/** list all versions along with their SHA digest */
+let versionsArr = []; const versionMap = {'v2Components': [], 'v3Components': [], 'v4Components': []}
+
+try {
+  versionsArr = JSON.parse(fs.readFileSync(process.env.VERSION_FILE));
```
Comment (arvindanta) [inline www/.vuepress/config.js:22]: can you also add a default value if process.env.VERSION_FILE is not set. will be useful when it is run from local

File: www/scripts/addversion.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+      .readFileSync(
+        path.join(rootPath, 'packages/crayons-core', 'package.json')
+      )
+      .toString();
+
+    const value = JSON.parse(file);
+
+    const key = value.version;
+
+    const versionsArr = JSON.parse(fs.readFileSync(process.env.VERSION_FILE));
```
Comment (arvindanta) [inline www/scripts/addversion.js:49]: can you also add a default value if process.env.VERSION_FILE is not set

**Review by arvindanta — CHANGES_REQUESTED** [review]

File: www/.vuepress/config.js (hunk 3 above)
Comment (srivalli-sivaramasankaran) [reply www/.vuepress/config.js:33]: added

File: www/scripts/addversion.js (hunk 4 above)
Comment (srivalli-sivaramasankaran) [reply www/scripts/addversion.js:60]: added.

File: www/scripts/template.md (hunk 2 above)
Comment (srivalli-sivaramasankaran) [reply www/scripts/template.md]: done

File: www/.vuepress/config.js (hunk 1 above)
Comment (srivalli-sivaramasankaran) [reply www/.vuepress/config.js]: fixed.

File: www/.vuepress/config.js [hunk 5]
```diff
@@ -15,6 +15,34 @@ const customObjectsComponents = components.filter((comp) =>
...
+} catch (err) {
+  console.log('version.json fetch error ', err);
+  versionArr = [];
+}
+versionsArr.forEach((v) => {
+  v['key'] === '2.13.4' && versionMap['v2Components'].push(`versions/v${v['key']?.split('.')[0]}.x/${v.key}/`);
```
[the remark this answers is not in the document]
Comment (srivalli-sivaramasankaran) [reply www/.vuepress/config.js]: done.

**Review by arvindanta — APPROVED** [review]

**Pull request merged**